ibis/pandas/execution/selection.py

```diff
@@ -107,16 +118,25 @@ def compute_projection_column_expr(
         )
 
     data_columns = frozenset(data.columns)
-    additional_scope = {
-        t: map_new_column_names_to_data(
-            remap_overlapping_column_names(parent_table_op, t, data_columns),
-            data,
+
+    additional_scope = {}
+    for t in op.root_tables():
+        additional_scope = toolz.merge(
+            additional_scope,
+            make_scope_item(
```

Comment:
It's a little confusing that you sometime use 

```
toolz.merge(scope, another_scope)
```
and sometimes use:
```
set_scope_item(scope, another_scope)
```

Can you unify?

Reply:
I agree it's a little confusing if we want to distinguish merge(set multiple items) and set (set singe item). 
The change now is intended to do minimum changes to the original workflow. I will try unify these usages.

Reply:
I am not sure we need a method to set single item. Is `merge` enough?